src/box/key_def.h

```diff
@@ -88,6 +89,8 @@ struct key_part {
 	uint32_t coll_id;
 	/** Collation definition for string comparison */
 	struct coll *coll;
+	/** Data of pinned coll_id in collation identifier cache. */
+	struct coll_id_cache_holder coll_id_cache_holder;
```

Comment:
I'm not sure about keeping the holder in `struct key_part`. This looks like an overkill because this field is only used for `key_def` objects attach to a space as the primary key. Maybe, better store it in `struct space`? This would require an extra allocation though. Need to discuss with @alyapunov.